app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java

```diff
@@ -180,5 +213,79 @@ protected void onPreExecute() {
                     "");
         }
     }
+
+    /**
+     * Read the barcode in background and post the results on barcodeView.
+     * Documentation of AsyncTask at https://developer.android.com/reference/android/os/AsyncTask
+     *
+     * @author Luca Perali (g4) (inspired to AsyncLoad class)
+     */
+    private class BarcodeReader extends AsyncTask<Bitmap, Void, String>{
+
+        protected ProgressDialog progressDialog; //TODO: progressDialog is deprecated, replace with ProgressBar
+        protected TextView barcodeTextView; //
+        protected String progressMessage;
+
+        /**
+         * Costructor initialize:
+         * @param barcodeTextView the View responsable for barcode rapresentation
+         * @param progressMessage message of the progress bar
+         */
+        BarcodeReader(TextView barcodeTextView, String progressMessage){
+            this.barcodeTextView = barcodeTextView;
+            this.progressMessage = progressMessage;
+        }
+
+        @Override
+        protected void onPreExecute() {
+            progressDialog = ProgressDialog.show(ResultActivity.this, progressMessage, "Barcode detection...");
+        }
+
+        @Override
+        protected String doInBackground(Bitmap... bitmaps){
+            String rawBarcode = "";
+            //BarcodeExtractor barcodeExtractor = ;
+
+            if(lastPhoto != null) {
+                rawBarcode = new BarcodeExtractor().getTextFromImg(lastPhoto); //see BarcodeExtractor class
+
+                if(rawBarcode.equals("")) //No barcode found case
```

Comment:
In my opinion the brace should be put in the same line of the "if". Here a my suggestion
```suggestion
                //No barcode found case
                if(rawBarcode.equals("")){
```